js/models/questionModel.js

```diff
@@ -65,6 +65,7 @@ class QuestionModel extends ComponentModel {
   }
 
   init() {
+    this._contextActivities = [];
```

Comment:
If you add a type definition, just below the imports, at the top of the file, like this:
```js
/**
 * @typedef {Object} ContextActivity
 * @property {string} id
 * @property {string} type
 * @property {string} title
 */
```
Then you can refer to that type throughout:
```suggestion
    /** @type {[ContextActivity]} */
    this._contextActivities = [];
```

Then you'll get:
![image](https://github.com/user-attachments/assets/f26105d4-d134-4226-9e56-4f2fef81b76a)

Reply:
A class is heavy here for what is effectively just a structured object.